docs/app-check.rst

```diff
@@ -0,0 +1,79 @@
+##############
+App Check
+##############
+
+The Firebase Admin SDK for PHP provides an API for verifying custom backends using Firebase App Check.
+
+Before you start, please read about Firebase App Check in the official documentation:
+
+* `Introduction to the Admin Database API <https://firebase.google.com/docs/app-check>`_
+* `Verify App Check tokens from a custom backend <https://firebase.google.com/docs/app-check/custom-resource-backend>`_
+* `Implement a custom App Check provider <https://firebase.google.com/docs/app-check/custom-provider>`_
+
+*******************************
+Initializing the App Check component
+*******************************
+
+**With the SDK**
+
+.. code-block:: php
+
+    $appCheck = $factory->createAppCheck();
+
+**With Dependency Injection** (`Symfony Bundle <https://github.com/kreait/firebase-bundle>`_/`Laravel/Lumen Package <https://github.com/kreait/laravel-firebase>`_)
+
+.. code-block:: php
+
+    use Kreait\Firebase\Contract\AppCheck;
+
+    class MyService
+    {
+        public function __construct(AppCheck $appCheck)
+        {
+            $this->appCheck = $appCheck;
+        }
+    }
+
+**With the Laravel** ``app()`` **helper** (`Laravel/Lumen Package <https://github.com/kreait/laravel-firebase>`_)
+
+.. code-block:: php
+
+    $auth = app('firebase.app_check');
+
+
+.. _verify-app-check-tokens:
+
+***********************
+Verify App Check Tokens
+***********************
+
+The Firebase Admin SDK has a built-in method for validating App Check tokens.
+
+See https://firebase.google.com/docs/app-check/custom-resource-backend for more information.
+
+.. code-block:: php
+
+    use Kreait\Firebase\Exception\AppCheck\FailedToVerifyAppCheckToken;
+
+    $token = getallheaders()['X-Firebase-AppCheck'];
+
+    try {
+        $appCheck->verifyToken($token);
+    } catch (FailedToVerifyAppCheckToken $e) {
```

Comment:
I would document this the same way as, for example, the ID Token verification. Depending on the project, people might retrieve the AppCheck Token in different ways, `getallheaders()` is an implementation details which some users might interpret as "the thing to do" and then open issues when things don't work out 😅 

```suggestion
    $appCheckTokenString = '...';

    try {
        $appCheck->verifyToken($appCheckTokenString);
    } catch (FailedToVerifyAppCheckToken $e) {
```